shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/ManagedBridgeServiceImpl.java

```diff
@@ -97,34 +110,86 @@ private void ensureManagedBridgeLabels(ManagedBridge managedBridge, final HasMet
         }
     }
 
-    private void createOrUpdateBridgeSecret(ManagedBridge managedBridge, BridgeDTO bridgeDTO) {
-        Secret expected = templateProvider.loadBridgeIngressSecretTemplate(managedBridge, TemplateImportConfig.withDefaults(LabelsBuilder.V2_OPERATOR_NAME));
+    private void createOrUpdateKnativeBrokerConfigurationSecret(ManagedBridge managedBridge, KnativeBrokerConfigurationDTO knativeBrokerConfiguration) {
 
-        KafkaConnectionDTO kafkaConnection = bridgeDTO.getKnativeBrokerConfiguration().getKafkaConnection();
-        expected.getData().put(GlobalConfigurationsConstants.KNATIVE_KAFKA_BOOTSTRAP_SERVERS_SECRET,
-                Base64.getEncoder().encodeToString(kafkaConnection.getBootstrapServers().getBytes()));
-        expected.getData().put(GlobalConfigurationsConstants.KNATIVE_KAFKA_USER_SECRET, Base64.getEncoder().encodeToString(kafkaConnection.getClientId().getBytes()));
-        expected.getData().put(GlobalConfigurationsConstants.KNATIVE_KAFKA_PASSWORD_SECRET, Base64.getEncoder().encodeToString(kafkaConnection.getClientSecret().getBytes()));
+        Secret expected = templateProvider.loadBridgeIngressSecretTemplate(managedBridge, new TemplateImportConfig(V2_OPERATOR_NAME)
+                .withNamespaceFromParent()
+                .withOwnerReferencesFromParent());
+        expected.getMetadata().setName(KNATIVE_BROKER_SECRET_NAME);
```

Comment:
Why can this not be done in the `loadBridgeIngressSecretTemplate(...)` method? It seems strange that the name is set in the metadata _outside_ of the `TemplateProvider` that updates all other metadata. I'd rather `name` either be added to `TemplateImportConfig` (and not just the ability to set it from parent) or overload the `loadBridgeIngressSecretTemplate(...)` method.